envs/doom/multiplayer/doom_multiagent_wrapper.py

```diff
@@ -13,6 +14,31 @@
 from envs.doom.doom_render import concat_grid, cvt_doom_obs
 from envs.doom.multiplayer.doom_multiagent import find_available_port, DEFAULT_UDP_PORT
 from utils.utils import log
+from functools import wraps
+from time import sleep
+
+
+def retry_dm(exception_class=Exception, num_attempts=3, sleep_time=1):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
```

Comment:
I don't think I can pass extra arguments called `wrapper` and `should_reset`. This is because the only parameter passed to the function is `self` and that's the only thing I can read in the `*args` as `args[0]`

Reply:
The should be passed as function parameters to any of the functions and should be used within the function, for example within `step`. If I just pass it as a function parameter (only then can the decorator access it) and don't use it, makes no sense.

Reply:
I don't remember by heart how decorators work on class methods, but if that's the case please at least create a named variable for args[0] inside the function (call it multiagent_wrapper or something).
Should_reset you can pass as an argument to the decorator itself, it does not have to be the argument to the wrapper() function.

Reply:
Have made the changes, do verify it